File: Python/2015/Happy or Sad.py
# J2

str1 = input("Enter a string: ")

if len(str1) < 1 or len(str1) > 255:
    print("That is not a valid input.")
    exit()

happy = str1.count(":-)")
sad = str1.count(":-(")

if happy > sad:
    print("Happy")
elif happy < sad:
    print("Sad")
elif happy == -1 and sad == -1:
    print("None")
else:
    print("Unsure")

File: C++/2015/Happy or Sad.cpp
// J2

#include <iostream>
#include <vector>

using namespace std;

int countOccurences(string data, string toSearch) {
    int occurences = 0;
    size_t pos = data.find(toSearch);

    while(pos != string::npos) {
        occurences++;

        pos = data.find(toSearch, pos + toSearch.size());
    }

    return occurences;
}

int main() {
    string input;

    getline(cin, input);

    int happycount, sadcount;

    happycount = countOccurences(input, ":-)");
    sadcount = countOccurences(input, ":-(");

    if (happycount == 0 && sadcount == 0) cout << "none" << endl;
    else if (happycount > sadcount) cout << "happy" << endl;
    else if (happycount < sadcount) cout << "sad" << endl;
    else cout << "unsure" << endl;

        
}